python/version_details/version_details.py

```diff
@@ -586,7 +587,7 @@ def load_data(self, entity):
         self._requested_entity = entity
 
         # If we're pinned, then we don't allow loading new entities.
-        if self._pinned and self.current_entity:
+        if self._pinned and self._current_entity:
```

Comment:
I can see you've done #3 in at least one other pr so that's good - would be good to see your thoughts on #2.